jetty-hazelcast/src/main/java/org/eclipse/jetty/hazelcast/session/HazelcastSessionDataStore.java

```diff
@@ -0,0 +1,211 @@
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.hazelcast.session;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import org.eclipse.jetty.server.session.AbstractSessionDataStore;
+import org.eclipse.jetty.server.session.SessionContext;
+import org.eclipse.jetty.server.session.SessionData;
+import org.eclipse.jetty.server.session.SessionDataMap;
+import org.eclipse.jetty.server.session.SessionDataStore;
+import org.eclipse.jetty.util.annotation.ManagedObject;
+import org.eclipse.jetty.util.component.AbstractLifeCycle;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Session data stored in Hazelcast
+ */
+@ManagedObject
+public class HazelcastSessionDataStore
+    extends AbstractSessionDataStore
+    implements SessionDataStore
+{
+
+    private  final static Logger LOG = Log.getLogger( "org.eclipse.jetty.server.session");
+
+    private HazelcastInstance hazelcastInstance;
+
+    private String jettySessionMapName;
+
+    private IMap<String, SessionData> sessionDataMap;
+
+    public HazelcastSessionDataStore()
+    {
+        // no op
+    }
+
+    public HazelcastSessionDataStore( HazelcastInstance hazelcastInstance, String jettySessionMapName )
+    {
+        this.hazelcastInstance = hazelcastInstance;
+        this.jettySessionMapName = jettySessionMapName;
+        this.sessionDataMap = hazelcastInstance.getMap( getJettySessionMapName() );
+    }
+
+    public HazelcastInstance getHazelcastInstance()
+    {
+        return hazelcastInstance;
+    }
+
+    public void setHazelcastInstance( HazelcastInstance hazelcastInstance )
+    {
+        this.hazelcastInstance = hazelcastInstance;
+    }
+
+    public String getJettySessionMapName()
+    {
+        return jettySessionMapName;
+    }
+
+    public void setJettySessionMapName( String jettySessionMapName )
+    {
+        this.jettySessionMapName = jettySessionMapName;
+    }
+
+
+    @Override
+    public SessionData load( String id )
+        throws Exception
+    {
+        return sessionDataMap == null ? null : sessionDataMap.get( id );
+    }
+
+    @Override
+    public boolean delete( String id )
+        throws Exception
+    {
+        return sessionDataMap == null ? false : sessionDataMap.remove( id ) != null;
+    }
+
+    @Override
+    public void initialize( SessionContext context )
+        throws Exception
+    {
+        _context = context;
+        if (this.sessionDataMap == null)
+        {
+            this.sessionDataMap = getHazelcastInstance().getMap( getJettySessionMapName() );
+        }
+    }
+
+    @Override
+    public void store( String id, SessionData data )
```

Comment:
Why do you need to override the store method? Implementing doStore() should be enough.